sync/src/main/java/org/bf2/sync/SecretManager.java

```diff
@@ -0,0 +1,231 @@
+package org.bf2.sync;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import org.bf2.common.OperandUtils;
+import org.bf2.operator.resources.v1alpha1.*;
+import org.bf2.sync.informer.InformerManager;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+
+@ApplicationScoped
+public class SecretManager {
+
+    private static final String ENDPOINT_TLS_CRT = "endpoint.tls.crt";
+    private static final String ENDPOINT_TLS_KEY = "endpoint.tls.key";
+    private static final String OAUTH_SSO_CLIENT_ID = "oauth.ssoClientId";
+    private static final String OAUTH_SSO_CLIENT_SECRET = "oauth.ssoClientSecret";
+    private static final String CANARY_SASL_PRINCIPAL = "canary.sasl.principal";
+    private static final String CANARY_SASL_PASSWORD = "canary.sasl.password";
+    private static final String ANNOTATION_MASTER_SECRET_DIGEST = "managedkafka.bf2.org/master-secret-digest";
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    @Inject
+    InformerManager informerManager;
+
+    public static String masterSecretName(ManagedKafka managedKafka) {
+        return managedKafka.getMetadata().getName() + "-master-secret";
+    }
+
+    public static boolean isKafkaAuthenticationEnabled(ManagedKafka managedKafka) {
+        return managedKafka.getSpec().getOauth() != null;
+    }
+
+    public static boolean isKafkaExternalCertificateEnabled(ManagedKafka managedKafka) {
+        return managedKafka.getSpec().getEndpoint().getTls() != null;
+    }
+
+    public static boolean isCanaryServiceAccountPresent(ManagedKafka managedKafka){
+        return managedKafka.getServiceAccount(ServiceAccount.ServiceAccountName.Canary).isPresent();
+    }
+
+    public boolean masterSecretExists(ManagedKafka managedKafka){
```

Comment:
maybe this should be changed to `isMasterSecretExists` to meet the other methods nomenclature